tests/bluetooth/tester/src/btp_bap.c

```diff
@@ -17,35 +18,1264 @@
+			return stream;
+		}
+	}
+
+	return NULL;
+}
+
+static struct audio_stream *stream_find(struct audio_connection *conn, uint8_t ase_id)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(conn->streams); i++) {
+		struct bt_bap_stream *stream = &conn->streams[i].stream;
+
+		if (stream->ep != NULL && stream->ep->status.id == ase_id) {
+			return &conn->streams[i];
+		}
+	}
+
+	return NULL;
+}
+
+static struct bt_bap_ep *end_point_find(struct audio_connection *conn, int ase_id)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(conn->end_points); i++) {
+		struct bt_bap_ep *ep = conn->end_points[i];
+
+		if (ep->status.id == ase_id) {
+			return ep;
+		}
+	}
+
+	return NULL;
+}
+
+static void btp_send_ascs_operation_completed_ev(struct bt_conn *conn, uint8_t ase_id,
+						 uint8_t opcode, uint8_t status)
+{
+	struct btp_ascs_operation_completed_ev ev;
+	struct bt_conn_info info;
+
+	(void)bt_conn_get_info(conn, &info);
+	bt_addr_le_copy(&ev.address, info.le.dst);
+	ev.ase_id = ase_id;
+	ev.opcode = opcode;
+	ev.status = status;
+	ev.flags = 0;
+
+	tester_event(BTP_SERVICE_ID_ASCS, BTP_ASCS_EV_OPERATION_COMPLETED, &ev, sizeof(ev));
+}
+
+static int validate_codec_parameters(const struct bt_codec *codec)
+{
+	int freq_hz;
+	int frame_duration_us;
+	int frames_per_sdu;
+	int octets_per_frame;
+	int chan_allocation_err;
+	enum bt_audio_location chan_allocation;
+
+	freq_hz = bt_codec_cfg_get_freq(codec);
+	frame_duration_us = bt_codec_cfg_get_frame_duration_us(codec);
+	chan_allocation_err = bt_codec_cfg_get_chan_allocation_val(codec, &chan_allocation);
+	octets_per_frame = bt_codec_cfg_get_octets_per_frame(codec);
+	frames_per_sdu = bt_codec_cfg_get_frame_blocks_per_sdu(codec, true);
+
+	if (freq_hz < 0) {
+		LOG_DBG("Error: Invalid codec frequency.");
+		return errno;
+	}
+
+	if (frame_duration_us < 0) {
+		LOG_DBG("Error: Invalid frame duration.");
+		return errno;
+	}
+
+	if (octets_per_frame < 0) {
+		LOG_DBG("Error: Invalid octets per frame.");
+		return errno;
+	}
+
+	if (frames_per_sdu < 0) {
+		LOG_DBG("Error: Invalid frames per sdu.");
+		return errno;
+	}
+
+	if (chan_allocation_err < 0) {
+		LOG_DBG("Error: Invalid channel allocation.");
+		return errno;
+	}
+
+	return 0;
+}
+
+static int lc3_config(struct bt_conn *conn, const struct bt_bap_ep *ep, enum bt_audio_dir dir,
+		      const struct bt_codec *codec, struct bt_bap_stream **stream,
+		      struct bt_codec_qos_pref *const pref, struct bt_bap_ascs_rsp *rsp)
+{
+	struct audio_connection *audio_conn;
+	struct audio_stream *stream_wrap;
+
+	LOG_DBG("ASE Codec Config: ep %p dir %u", ep, dir);
+
+	print_codec(codec);
+
+	if (validate_codec_parameters(codec)) {
+		*rsp = BT_BAP_ASCS_RSP(BT_BAP_ASCS_RSP_CODE_CONF_REJECTED,
+				       BT_BAP_ASCS_REASON_CODEC_DATA);
+
+		btp_send_ascs_operation_completed_ev(conn, ep->status.id, BT_ASCS_CONFIG_OP,
+						     BTP_ASCS_STATUS_FAILED);
+
+		return -ENOTSUP;
+	}
+
+	audio_conn = find_connection(conn);
+	stream_wrap = stream_alloc(audio_conn);
```

Comment:
audio_conn should be check for null before calling stream_alloc()   (or asserted if it cannot be null here)